service/storage_service.cc

```diff
@@ -5939,6 +5939,35 @@ future<> storage_service::do_tablet_operation(locator::global_tablet_id tablet,
     }
 }
 
+future<> storage_service::repair_tablet(locator::global_tablet_id tablet) {
+    return do_tablet_operation(tablet, "Repair", [this, tablet] (locator::tablet_metadata_guard& guard) -> future<> {
+        slogger.debug("Executing repair for tablet={}", tablet);
+        auto tm = guard.get_token_metadata();
+        auto& tmap = guard.get_tablet_map();
+        auto* trinfo = tmap.get_tablet_transition_info(tablet.tablet);
+
+        // Check if the request is still valid.
+        // If there is mismatch, it means this repair was canceled and the coordinator moved on.
+        if (!trinfo) {
+            throw std::runtime_error(fmt::format("No transition info for tablet {}", tablet));
+        }
+        if (trinfo->stage != locator::tablet_transition_stage::repair) {
+            throw std::runtime_error(fmt::format("Tablet {} stage is not at repair", tablet));
+        }
+        if (trinfo->session_id) {
+            // TODO: Propagate session id to repair
+            slogger.debug("repair_tablet: tablet={} session_id={}", tablet, trinfo->session_id);
+        } else {
+            throw std::runtime_error(fmt::format("Tablet {} session is not set", tablet));
+        }
+
+        utils::get_local_injector().inject("repair_tablet_fail_on_rpc_call",
+            [] { throw std::runtime_error("repair_tablet failed due to error injection"); });
+        co_await _repair.local().repair_tablet(guard, tablet);
```

Comment:
I will drop the tm pointer here. However, repair will hold erm internally until repair finishes. Resetting tm here is not going to allow new topology changes. We will work on the allow migration and bootstrap in parallel in another PR.

Reply:
> We will work on the allow migration and bootstrap in parallel in another PR.

It won't help with parallelism for tablet operations, which also rely on barriers.

Can we switch tablet repair to use tablet_metadata_guard, which doesn't block erm if the change is unrelated to the associated tablet?

Reply:
We already use the tablet_metadata_guard for the tablet repair part of the code. However the generic repair code holds erm in may places since the erm thing is introduced. We need a separate project to address this.